src/mod_amp.erl

```diff
@@ -59,7 +59,7 @@ run_initial_check(Acc, _C2SState) ->
 
 
 -spec check_packet(mongoose_acc:t() | exml:element(), amp_event()) ->
-    mongoose_acc:t() | exml:element() | drop.
+    mongoose_acc:t() | exml:element().
```

Comment:
isn't it possible to drop the packet at this point?

Reply:
It is possible. As far as I know the handlers may add this decision to the accumulator and the hook caller can check the decision stored in the Acc. When it comes to `mod_amp`, as far as I know it was never dropping the packet at this point, only adding some info to the packet. This is pre Acc solution which I know needs to be reworked to better utilize the Acc.